### ISSUE TITLE
[openbmc] rpower state command should check Transition and HostState values.

### ISSUE THREAD

**Issue by gurevichmark** [xcat2/xcat-core#3014]
Opening this issue bases on the observation reported in PR #2978

1. It seem with the current p9 hardware it takes about 13 seconds for the machine to power on.

The `rpower <> stat` command should examine the values of `RequestedHostTransition` and `CurrentHostState`.
If both are "Off" or "On", the state returned from the command should be "off" or "on" respectively.

But if the values are as following, the state returned from the command should be "powering-on"
```
'RequestedHostTransition' => 'xyz.openbmc_project.State.Host.Transition.On',
'CurrentHostState' => 'xyz.openbmc_project.State.Host.HostState.Off'
```

For example, after issuing `rpower <> on`, the following was observed:
```
Tue May  9 13:45:39 EDT 2017
p9euh01: DEBUG host_states RequestedHostTransition=xyz.openbmc_project.State.Host.Transition.On
p9euh01: DEBUG host_states CurrentHostState=xyz.openbmc_project.State.Host.HostState.Off
Tue May  9 13:45:42 EDT 2017
p9euh01: DEBUG host_states RequestedHostTransition=xyz.openbmc_project.State.Host.Transition.On
p9euh01: DEBUG host_states CurrentHostState=xyz.openbmc_project.State.Host.HostState.Off
Tue May  9 13:45:45 EDT 2017
p9euh01: DEBUG host_states RequestedHostTransition=xyz.openbmc_project.State.Host.Transition.On
p9euh01: DEBUG host_states CurrentHostState=xyz.openbmc_project.State.Host.HostState.Off
Tue May  9 13:45:52 EDT 2017
p9euh01: DEBUG host_states RequestedHostTransition=xyz.openbmc_project.State.Host.Transition.On
p9euh01: DEBUG host_states CurrentHostState=xyz.openbmc_project.State.Host.HostState.Running
Tue May  9 13:45:55 EDT 2017
p9euh01: DEBUG host_states RequestedHostTransition=xyz.openbmc_project.State.Host.Transition.On
p9euh01: DEBUG host_states CurrentHostState=xyz.openbmc_project.State.Host.HostState.Running
```

2. There is a mismatch between the state that `rpower <> stat` returns and the value of `status` attribute in node definition:
```
[root@stratton01 xCAT_plugin]# lsdef p9euh01 -i status -c
p9euh01: status=powering-off
[root@stratton01 xCAT_plugin]# rpower p9euh01 stat
Warning: Currently running development code, use at your own risk.  Unset XCAT_OPENBMC_DEVEL to disable.
p9euh01: off
[root@stratton01 xCAT_plugin]#
```
and
```
[root@stratton01 xCAT_plugin]# lsdef p9euh01 -i status -c
p9euh01: status=powering-on
[root@stratton01 xCAT_plugin]# rpower p9euh01 stat
Warning: Currently running development code, use at your own risk.  Unset XCAT_OPENBMC_DEVEL to disable.
p9euh01: on
[root@stratton01 xCAT_plugin]#
```

3. Anything needs to be done for `nodestat` command ?

**Comment by whowutwut** [comment]
@gurevichmark Do you want to work on this issue?

**Comment by gurevichmark** [comment]
Fixed by PR #3104
